services/server/server_test.go

```diff
@@ -86,48 +86,50 @@ func TestServer(t *testing.T) {
 			err := s.UpdateSettingsFromEnv([]string{
 				"DISABLE_TELEMETRY",
 			})
-			require.NoError(t, err)
-			assert.Equal(t, false, s.envDisableTelemetry)
+			require.Errorf(t, err, "validation error of environment variables")
```

Comment:
Let's check for specific error message that should include environment variable name.